### PR TITLE
fix: Disable creating plays in Production

### LINKED ISSUE #420: 🐛 [Bug report]: Can create a play in production
**Describe the bug**
I'm allowed to create a play in production

**To Reproduce**
1. Go to [reactplay.io/plays/create](https://reactplay.io/plays/create)

**Expected behavior**
I mustn't be able create a play in production

**Screenshots**
image

### PR DESCRIPTION
**Pull request by 6km**
# Description

Fixes #420

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by atapas** [comment]
@koustov please finish review.

File: src/common/routing/RouteDefs.jsx [hunk 1]
```diff
@@ -36,8 +36,12 @@ const RouteDefs = () => {
                 <PlayList/>
               }
             />
-                  <Route exact path="create" element= {<CreatePlay />}/>
-                  <Route exact path="created/:playid" element= {<PlayCreated />}/>
+                  { process.env.NODE_ENV === "development"
+                    && <>
+                      <Route exact path="create" element= {<CreatePlay />}/>
```
Comment (koustov) [inline src/common/routing/RouteDefs.jsx:41]: Hey @6km . I would suggest having the path intact however when someone hits this path on production the page will show a custom message something like `You can only create a play in dev mode. For details visit <md_link>`. This is applicable only for `create` route only.

On the other hand, `created/:playid` will continue to be wrapped withing the "if" condition

**Review by koustov — CHANGES_REQUESTED** [review]
A comment dropped

File: src/common/routing/RouteDefs.jsx (hunk 1 above)
Comment (6km) [reply src/common/routing/RouteDefs.jsx]: @koustov done, can you re-review please?

**Comment by atapas** [comment]
@6km Code changes look good to me. Only one comment, the warning image added is not aligned with the app theme. Here is an example: https://github.com/reactplay/react-play/blob/main/src/images/img-404.svg

**Comment by 6km** [comment]
> @6km Code changes look good to me. Only one comment, the warning image added is not aligned with the app theme. Here is an example: [`main`/src/images/img-404.svg](https://github.com/reactplay/react-play/blob/main/src/images/img-404.svg?rgh-link-date=2022-08-01T06%3A58%3A04Z)

@atapas I changed the image with a copy of img-404.svg but it shows 403 instead.
image

File: src/common/createplay/CreatePlay.jsx [hunk 2]
```diff
@@ -11,6 +11,21 @@ import Loader from "common/spinner/spinner";
...
+const NoCreationInProdScreen = () => {
+  return <div className="w-full h-full flex flex-col justify-center items-center gap-8">
+    <div>
+      <NotAllowedImage width={550} />
+    </div>
+
+    <div className="text-center">
+      You can't create plays in production
+      <br />
+      <a className="text-link-default" href="https://github.com/reactplay/react-play#readme" target="__blank__">read this</a> for more details
```
Comment (atapas) [inline src/common/createplay/CreatePlay.jsx:25]: Shouldn't it point to the create play md? CREATE-PLAY.md

Comment (6km) [reply src/common/createplay/CreatePlay.jsx]: done changed it 👍

File: src/common/createplay/CreatePlay.jsx [hunk 3]
```diff
@@ -11,6 +11,21 @@ import Loader from "common/spinner/spinner";
...
+const NoCreationInProdScreen = () => {
+  return <div className="w-full h-full flex flex-col justify-center items-center gap-8">
+    <div>
+      <NotAllowedImage width={550} />
+    </div>
+
+    <div className="text-center">
+      You can't create plays in production
+      <br />
+      <a className="text-link-default" href="https://github.com/reactplay/react-play/blob/main/CREATE-PLAY.md" target="__blank__">read this</a> for more details
```
Comment (koustov) [inline src/common/createplay/CreatePlay.jsx:25]: This is interesting, what is `__blank__` instead of `_blank`?

Comment (koustov) [reply src/common/createplay/CreatePlay.jsx:25]: Doesn't it give any warning as you are not using `rel` property?

Comment (6km) [reply src/common/createplay/CreatePlay.jsx]: @koustov After a little searching, I found that `__blank__` isn't a valid target value the true one is `_blank`, and also found that the `rel` attribute isn't required in major browsers as it's automatically set to `rel="noopener"`.

- [Source](https://stackoverflow.com/a/50709724/14088437)

Comment (6km) [reply src/common/createplay/CreatePlay.jsx]: Btw, I have changed the attributes to `target="_blank"` and `rel="noopener noreferrer"`.

Comment (koustov) [reply src/common/createplay/CreatePlay.jsx]: > @koustov After a little searching, I found that `__blank__` isn't a valid target value the true one is `_blank`, and also found that the `rel` attribute isn't required in major browsers as it's automatically set to `rel="noopener"`.
> 
> * [Source](https://stackoverflow.com/a/50709724/14088437)

I was so curious :)

**Review by koustov — DISMISSED** [review]
lgtm

**Comment by atapas** [comment]
@6km please resolve conflicts

**Comment by 6km** [comment]
> @6km please resolve conflicts

done 👌

**Review by koustov — APPROVED** [review]
lgtm

**Pull request merged**